### PR TITLE
add rosdep key for geomag

### PR DESCRIPTION
**Pull request by tdenewiler**
Used to find magnetic declination for any given latitude/longitude/altitude. Allows easier calculation of global heading from IMU or compass data in robots distributed around the world. This package is only available via PyPI, not OS package managers.

* PyPI: https://pypi.org/project/geomag/
* Arch: https://www.archlinux.org/packages/?sort=&q=geomag&maintainer=&flagged=
* Debian: https://packages.debian.org/search?suite=all&section=all&arch=any&searchon=names&keywords=geomag
* Fedora: https://apps.fedoraproject.org/packages/s/geomag
* Gentoo: https://packages.gentoo.org/packages/search?q=geomag
* Ubuntu: https://packages.ubuntu.com/search?suite=all&section=all&arch=any&keywords=geomag&searchon=names

### CODE REVIEW

**Review by clalancette — APPROVED** [review]

File: rosdep/python.yaml [hunk 1]
```diff
@@ -149,6 +149,19 @@ exhale-pip:
   ubuntu:
     pip:
       packages: [exhale]
+geomag:
```
Comment (sloretz) [inline rosdep/python.yaml:152]: Mind using the key name `geomag-pip`?

https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#pip-1

**Review by sloretz — CHANGES_REQUESTED** [review]

File: rosdep/python.yaml (hunk 1 above)
Comment (sloretz) [reply rosdep/python.yaml]: Also I see the contributing guide asks for `python-` and `python3-` suffixes. I guess that means this should be `python-geomag-pip` and `python3-geomag-pip`.

 @ros/rosdeputies Why is that the rule? Whether it's Python 2 or Python 3 depends on the version of python set in `ROS_PYTHON_VERSION` or the default rosdistro, what's the value of separate `python-` and `python3-` prefixes for keys only on PyPI?

Comment (tdenewiler) [reply rosdep/python.yaml]: c1858da is where I add `-pip` suffix. Let me know what you want for a prefix and I'll add it. I have the same questions you do about the prefix.

Comment (sloretz) [reply rosdep/python.yaml]: Opening PR to change recommendation: https://github.com/ros/rosdistro/pull/25508

Comment (clalancette) [reply rosdep/python.yaml]: I'm sorry for the back and forth on this, but in #25508 we eventually landed on putting the `python3-` prefix on the front and the `-pip` suffix on the back.  @tdenewiler Would you mind updating one more time to conform to that?  Then I think we're good to go here.

Comment (tdenewiler) [reply rosdep/python.yaml]: That was a good discussion in #25508, and this approach makes sense to me. Package management is not easy, I appreciate the thought going into this. I made the requested updates.

Comment (clalancette) [reply rosdep/python.yaml]: > That was a good discussion in #25508, and this approach makes sense to me. Package management is not easy, I appreciate the thought going into this. I made the requested updates.

Perfect, thanks for your patience.

**Review by clalancette — APPROVED** [review]

**Review by claireyywang — APPROVED** [review]

**Pull request merged**